numpy/lib/function_base.py

```diff
@@ -4246,46 +4339,53 @@ def percentile(a, q, axis=None, out=None,
 
     Notes
     -----
-    Given a vector ``V`` of length ``N``, the ``q``-th percentile of
-    ``V`` is the value ``q/100`` of the way from the minimum to the
+    Given a vector ``V`` of length ``N``, the ``q``-th quantile of
+    ``V`` is the value ``q`` of the way from the minimum to the
     maximum in a sorted copy of ``V``. The values and distances of
     the two nearest neighbors as well as the `interpolation` parameter
-    will determine the percentile if the normalized ranking does not
+    will determine the quantile if the normalized ranking does not
     match the location of ``q`` exactly. This function is the same as
-    the median if ``q=50``, the same as the minimum if ``q=0`` and the
-    same as the maximum if ``q=100``.
+    the median if ``q=0.5``, the same as the minimum if ``q=0.0`` and the
+    same as the maximum if ``q=1.0``.
 
     Examples
     --------
     >>> a = np.array([[10, 7, 4], [3, 2, 1]])
     >>> a
     array([[10,  7,  4],
            [ 3,  2,  1]])
-    >>> np.percentile(a, 50)
+    >>> np.quantile(a, 0.5)
     3.5
-    >>> np.percentile(a, 50, axis=0)
+    >>> np.quantile(a, 0.5, axis=0)
     array([[ 6.5,  4.5,  2.5]])
-    >>> np.percentile(a, 50, axis=1)
+    >>> np.quantile(a, 0.5, axis=1)
     array([ 7.,  2.])
-    >>> np.percentile(a, 50, axis=1, keepdims=True)
+    >>> np.quantile(a, 0.5, axis=1, keepdims=True)
     array([[ 7.],
            [ 2.]])
 
-    >>> m = np.percentile(a, 50, axis=0)
+    >>> m = np.quantile(a, 0.5, axis=0)
     >>> out = np.zeros_like(m)
-    >>> np.percentile(a, 50, axis=0, out=out)
+    >>> np.quantile(a, 0.5, axis=0, out=out)
     array([[ 6.5,  4.5,  2.5]])
     >>> m
     array([[ 6.5,  4.5,  2.5]])
 
     >>> b = a.copy()
-    >>> np.percentile(b, 50, axis=1, overwrite_input=True)
+    >>> np.quantile(b, 0.5, axis=1, overwrite_input=True)
     array([ 7.,  2.])
     >>> assert not np.all(a == b)
 
+    See Also
+    --------
+    percentile
+
     """
-    q = array(q, dtype=np.float64, copy=True)
-    r, k = _ureduce(a, func=_percentile, q=q, axis=axis, out=out,
+    q = asarray(q, dtype=np.float64)
+    if not _valid_quantile(q):
+        raise ValueError("Quantiles must be in the range [0.0, 1.0], "
+                         "or percentiles must be in hte range [0.0, 100.0]")
```

Comment:
I'd prefer it if these errors were separate as before. Should be easy with one more layer of helper function, with the (reversed) call graph looking like

* `_quantile` (rename to `_quantile_ureduce_func`)
   * `_quantile_unchecked` - expects a true `ndarray` in `[0, 1]`)
      * `quantile`
      * `percentile`